Api/Services/RestaurantService.cs

```diff
@@ -1407,5 +1422,55 @@ public async Task<Result<RestaurantVM>> GetRestaurantByIdAsync(int restaurantId)
 
             return result;
         }
+
+        public async Task<Result<Pagination<IngredientVM>>> GetIngredientsAsync(
+            int restaurantId,
+            IngredientSorting orderBy,
+            int page,
+            int perPage)
+        {
+            var restaurant = await context.Restaurants.FindAsync(restaurantId);
+            if (restaurant == null)
+            {
+                return new ValidationFailure
+                {
+                    PropertyName = null,
+                    ErrorMessage = $"Restaurant with ID {restaurantId} not found",
+                    ErrorCode = ErrorCodes.NotFound
+                };
+            }
+
+            IQueryable<Ingredient> query = context.MenuItems
+                .AsSplitQuery()
+                .Include(mi => mi.Ingredients)
+                .ThenInclude(imi => imi.Ingredient)
+                .Where(mi => mi.RestaurantId == restaurantId)
+                .SelectMany(mi => mi.Ingredients.Select(imi => imi.Ingredient))
+                .Distinct();
+
+            // Sortowanie
+            query = orderBy switch
+            {
+                IngredientSorting.NameAsc => query.OrderBy(i => i.PublicName),
+                IngredientSorting.NameDesc => query.OrderByDescending(i => i.PublicName),
+                IngredientSorting.AmountAsc => query.OrderBy(i => i.Amount),
+                IngredientSorting.AmountDesc => query.OrderByDescending(i => i.Amount),
+                _ => query.OrderBy(i => i.PublicName) // Domyślne sortowanie
```

Comment:
tu lepiej by było rzucić wyjątek, jedyny przypadek w którym uruchomiłaby się ta gałąź, to gdybyśmy dodali nową opcję sortowania, a zapomnielibyśmy dodać ją tu